src/main/java/uk/org/tombolo/importer/phe/AdultObesityImporter.java

```diff
@@ -61,65 +57,165 @@ public DatasourceSpec getDatasourceSpec(String datasourceIdString) throws Except
+                    continue;
+                }
+            }
+            // We need to check for again for invalid rows when looping through the local authorities names
+            catch (ArrayIndexOutOfBoundsException npe){
+                log.warn("Found invalid local authority row: Skipping");
+                continue;
+            }
         }
-        excelUtils.extractAndSaveTimedValues(sheet, this, timedValueExtractors);
+        // Finally we save the values in the database
+        saveAndClearTimedValueBuffer(timedValues);
     }
 
     @Override
     public List<Attribute> getTimedValueAttributes(String datasourceId) {
+
+        // Creating a placeholder for our attributes
         List<Attribute> attributes = new ArrayList<>();
-        attributes.add(new Attribute(getProvider(), AttributeLabel.fractionUnderweight.name(),
-                "BMI less than 18.5kg/m2"));
-        attributes.add(new Attribute(getProvider(), AttributeLabel.fractionHealthyWeight.name(),
-                "BMI greater than or equal to 18.5 but less than 25kg/m2"));
-        attributes.add(new Attribute(getProvider(), AttributeLabel.fractionOverweight.name(),
-                "BMI greater than or equal to 25 but less than 30kg/m2"));
-        attributes.add(new Attribute(getProvider(), AttributeLabel.fractionObese.name(),
-                "BMI greater than or equal to 30kg/m2"));
-        attributes.add(new Attribute(getProvider(), AttributeLabel.fractionExcessWeight.name(),
-                "BMI greater than or equal to 25kg/m2 (overweight including obese)"));
-        return attributes;
-    }
 
-    private int getAttributeColumnId(AttributeLabel attributeLabel){
-        switch (attributeLabel){
-           case fractionUnderweight:
-               return 6;
-            case fractionHealthyWeight:
-                return 10;
-            case fractionOverweight:
-                return 14;
-            case fractionObese:
-                return 18;
-            case fractionExcessWeight:
-                return 22;
-            default:
-                throw new Error("Unknown attribute label: " + String.valueOf(attributeLabel));
+        // Dataset specific: we hardcode the columns names for the our .csv file
+        String[] elements = { "fractionUnderweight", "fractionHealthyWeight", "fractionOverweight", "fractionObese", "fractionExcessWeight"};
+        String[] descriptions = { "BMI less than 18.5kg/m2", "BMI greater than or equal to 18.5 but less than 25kg/m2", "BMI greater than or equal to 25 but less than 30kg/m2", "BMI greater than or equal to 30kg/m2", "BMI greater than or equal to 25kg/m2 (overweight including obese)"};
+
+        // We loop through the elements of the elements object and adding an Attribute object in the list
+        // with nour column names.
+        for( int i = 0; i < elements.length; i++) {
+            attributes.add(new Attribute(getProvider(), elements[i], descriptions[i]));
```

Comment:
Nice to see use of ```arrays``` over ```arrayList```, great choice when you know the length of your elements 👍